packages/uploadthing/src/sdk/index.ts

```diff
@@ -55,8 +60,39 @@ export const getFileUrls = async (fileKeys: string[] | string) => {
     },
     body: JSON.stringify({ fileKeys }),
   });
+
   if (!res.ok) {
     throw new Error("Failed to get file urls");
   }
+
   return res.json().then(({ data }) => data as { key: string; url: string }[]);
 };
+
+/**
+ * Request file list from UploadThing storage.
+ *
+ * @example
+ * const data = await listFiles();
+ * console.log(data); // { key: "2e0fdb64-9957-4262-8e45-f372ba903ac8_image.jpg", id: "2e0fdb64-9957-4262-8e45-f372ba903ac8" }
+ */
+export const listFiles = async () => {
+  if (!UT_SECRET) throw new Error("Missing UPLOADTHING_SECRET env variable.");
+
+  // TODO: Implement filtering and pagination
+  const res = await fetch(generateUploadThingURL("/api/listFiles"), {
+    method: "POST",
+    headers: {
+      "Content-Type": "application/json",
+      "x-uploadthing-api-key": UT_SECRET,
+      "x-uploadthing-version": UPLOADTHING_VERSION,
+    },
+  });
+
+  if (!res.ok) {
+    throw new Error("Failed to list files");
```

Comment:
this error should probably be improved.

what i did for the server uploads was the following:

```ts
  const json = (await res.json()) as
    | { data: { key: string; url: string }[] }
    | { error: string };

  if (!res.ok || "error" in json) {
    const message = "error" in json ? json.error : "Unknown error";
    throw new Error(message);
  }

  return json.data;
```

Reply:
would obviously require a follow up PR to the infra

Reply:
Can do!